### PR TITLE
feat(client): UI for multiple providers

### PR DESCRIPTION
**Pull request by abrl91**
Close: #5812

## PR Details
 change the UI to support multiple providers

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by abrl91** [comment]
Leftovers:
As we can see in the screenshot below -

1. after a connection is established, the state of the git connection status in the footer is not updated
2. the selected group name is probably not saved after a creation/connection to a git repository

Screenshot 2023-05-05 at 19 04 26

File: packages/amplication-git-utils/src/providers/bitbucket/bitbucket.service.spec.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ describe("bitbucket.service", () => {
           path: "tests/",
           owner: "mr-bucket",
           repositoryName: "my-repo",
-          repositoryGroupName: "my-group",
+          groupName: "my-group",
```
Comment (overbit) [inline packages/amplication-git-utils/src/providers/bitbucket/bitbucket.service.spec.ts:90]: Always prefer consistency:
either `repositoryName` and `repositoryGroupName` or `name` and `groupName`

File: packages/amplication-git-utils/src/providers/bitbucket/bitbucket.service.spec.ts [hunk 2]
```diff
@@ -139,7 +139,7 @@ describe("bitbucket.service", () => {
         path: "tests/__init__.py",
         owner: "mr-bucket",
         repositoryName: "my-repo",
-        repositoryGroupName: "my-group",
+        groupName: "my-group",
```
Comment (overbit) [inline packages/amplication-git-utils/src/providers/bitbucket/bitbucket.service.spec.ts:142]: Always prefer consistency:
either `repositoryName` and `repositoryGroupName` or `name` and `groupName`

File: packages/amplication-client/src/Resource/git/SyncWithGithubPage.tsx [hunk 3]
```diff
@@ -35,7 +35,7 @@ const SyncWithGithubPage: React.FC = () => {
     },
     skip: !currentResource?.id,
   });
-
+  console.log("data", data);
```
Comment (overbit) [inline packages/amplication-client/src/Resource/git/SyncWithGithubPage.tsx:38]: Do we need this?

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/util/git-git-repository-details.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    provider === EnumGitProvider.Github
+      ? `${organizationName}/${repositoryName}`
+      : provider === EnumGitProvider.Bitbucket
+      ? `${groupName}/${repositoryName}`
+      : null;
+
+  const gitRepositoryUrlMap = {
+    [EnumGitProvider.Github]: `${GITHUB_URL}/${repositoryFullName}`,
+    [EnumGitProvider.Bitbucket]: `${BITBUCKET_URL}/${repositoryFullName}/src/master/`,
+  };
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-client/src/util/git-git-repository-details.ts]: There are some places where you still don't have gitOrganization/gitRepository/groupName - so yes - if one of them is not provided, you will get an invalid URL and probably a not found error. 
Do you think we need to add some checks here?

File: packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateGithubSync.tsx [hunk 5]
```diff
@@ -28,11 +28,12 @@ const CreateGithubSync: React.FC<props> = ({
...
   const gitProvider = gitRepository?.gitOrganization?.provider;
-  const gitRepositoryFullName = `${gitRepository?.gitOrganization?.name}/${gitRepository?.name}`;
-  const gitRepositoryUrl = getGitRepositoryUrlForServiceWizard(
+  const gitRepositoryUrl = getGitRepositoryDetails(
     gitProvider,
-    gitRepositoryFullName
-  );
+    gitRepository?.gitOrganization?.name,
+    gitRepository?.groupName,
+    gitRepository?.name
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateGithubSync.tsx:35]: I don't understand. These are not optional parameters

File: packages/amplication-server/src/core/git/git.provider.service.ts [hunk 6]
```diff
@@ -189,7 +189,7 @@ export class GitProviderService {
         type: EnumGitOrganizationType[organization.type],
         useGroupingForRepositories: organization.useGroupingForRepositories,
       },
-      repositoryGroupName: args.repositoryGroupName,
+      groupName: args.groupName,
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-server/src/core/git/git.provider.service.ts:195]: I go with name and groupName with all "Repository" related stuff

File: packages/amplication-server/src/core/git/git.provider.service.ts [hunk 7]
```diff
@@ -228,7 +228,7 @@ export class GitProviderService {
...
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-server/src/core/git/git.provider.service.ts:234]: I go with name and groupName with all "Repository" related stuff

**Review by overbit — APPROVED** [review]

File: packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateGithubSync.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateGithubSync.tsx]: I'll create an issue about it

**Pull request merged**